extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -236,6 +224,19 @@ export class AmpStoryPage extends AMP.BaseElement {
 
     /** @private {?number} Time at which an audio element failed playing. */
     this.playAudioElementFromTimestamp_ = null;
+
+    /**
+     * @return {!Promise<./media-pool.MediaPool>}
+     * @private
+     */
+    this.getPool_ = once(() => {
+      const storyElement = devAssert(
+        closestAncestorElementBySelector(this.element, 'amp-story')
+      );
+      return whenUpgradedToCustomElement(storyElement)
+        .then(() => storyElement.getImpl())
+        .then((impl) => impl.pool);
```

Comment:
`.pool` is a new public member, instead of using `MediaPool.for()`. The result of the latter is scoped by the story element in practice, so it's easier to remove that layer of indirection.

Reply:
(Future systems other than `amp-story` can provide the `MediaPool` however they like.)